src/Symfony/Component/Security/Core/User/InMemoryUserProvider.php

```diff
@@ -51,11 +51,12 @@ public function __construct(array $users = [])
      */
     public function createUser(UserInterface $user)
     {
-        if (isset($this->users[strtolower($user->getUsername())])) {
+        $username = strtolower(method_exists($user, 'getUserIdentifier') ? $user->getUserIdentifier() : $user->getUsername());
```

Comment:
Can we rename the variable to `$identifier`?  And change the `name` attribute accordingly in `InMemoryFactory` from security-bundle?